src/middleware/updateValidation.js

```diff
@@ -0,0 +1,82 @@
+const bcrypt = require('bcrypt');
+const constants = require('../lib/constants');
+const UserModel = require('../models/user');
+
+// eslint-disable-next-line consistent-return
+const validateUpdate = (req, res, next) => {
+  const {
+    password,
+    passwordConfirmation,
+    address,
+    age,
+    phoneNumber,
+    username,
+  } = req.body;
+  const errorsArray = [];
+  if (username) {
```

Comment:
Here you should also check if the username is occupied. We don't want Mongoose to intervene.

![Screenshot from 2022-08-16 16-21-01](https://user-images.githubusercontent.com/102853969/184892259-238272e0-a601-4a93-954e-1c56ce319c88.png)

We should get something like `This username is already in use`, and not the Mongoose error.

Reply:
Same thing with phone number.

Reply:
Thanks for checking Adnan, It's fixed now